# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/media/mediasink/DecodedStream.cpp
# Commit: 35d86243cb45
# Full Hash: 35d86243cb45f7e02dd6ec0eea04cd62c2a70196
# Author: Andrew Osmond <aosmond@mozilla.com>
# Date: 2024-09-16 09:14:26
# Regressor Bug: 1908245
# File Overlap Count: 1
# Description:
#   Bug 1908245 - Add support for missing requestVideoFrameCallback parameters. r=media-playback-reviewers,webidl,padenot,saschanaz
#   
#   This patch adds support for exposing the capture time, receive time,
#   processing duration and RTP timestamp parameters on the callback for
#   rVFC for WebRTC related elements.
# ==============================================================================

diff -r 7d143d92bf2a -r 35d86243cb45 dom/media/mediasink/DecodedStream.cpp
--- a/dom/media/mediasink/DecodedStream.cpp	Sun Sep 15 20:03:40 2024 +0000
+++ b/dom/media/mediasink/DecodedStream.cpp	Sun Sep 15 21:49:38 2024 +0000
@@ -847,7 +847,7 @@
     double aPlaybackRate) {
   RefPtr<layers::Image> image = aImage;
   aOutput->AppendFrame(image.forget(), aIntrinsicSize, aPrincipalHandle, false,
-                       aTimeStamp);
+                       aTimeStamp, media::TimeUnit::Invalid(), aStart);
   // Extend this so we get accurate durations for all frames.
   // Because this track is pushed, we need durations so the graph can track
   // when playout of the track has finished.